src/content/reference/react/cache.md

```diff
@@ -0,0 +1,443 @@
+---
+title: cache
+canary: true
+---
+
+<Canary>
+`cache` is currently only available in React’s [Canary](https://react.dev/community/versioning-policy#canary-channel) and [experimental](https://react.dev/community/versioning-policy#experimental-channel) channels. Please ensure you understand the limitations before using `cache` in production.
+
+Learn more about [React's release channels here](/community/versioning-policy#all-release-channels).
+
+</Canary>
+
+<Intro>
+
+`cache` lets you cache the result of a data fetch or computation.
+
+```js
+const cachedFn = cache(fn);
+```
+
+</Intro>
+
+<InlineToc />
+
+---
+
+## Reference {/*reference*/}
+
+### `cache(fn)` {/*cache*/}
+
+Wrap a function with `cache` to receive a version of that function with caching.
+
+```js {4,7}
+import {cache} from 'react';
+import fetchUser from 'lib/user';
+
+const getUser = cache(fetchUser);
+
+function UserProfile(id) {
+  const user = getUser(id);
+  //...
+}
+```
+
+`getUser` will first check if there is a cached result for `id`, otherwise call `fetchUser(id)` and cache the returned result.
```

Comment:
Is it intentionally implied that `fetchUser` returns a `user` value`?

My expectation would be for a function named `fetchUser` to be async (and return a `Promise`). Therefore, I would expect `getUser` to also be async (and return a `Promise`).

And if this example *is* implying that they are both async, I might be confused why the example does not show:

```
const user = await getUser(id);
```

Reply:
Updated!